draft-cms-masque-connect-ip.md

```diff
@@ -138,6 +138,33 @@ SETTINGS frame was received without the H3_DATAGRAM SETTINGS Parameter), the
 client MUST consider its CONNECT-IP request as failed.
 
 
+# Forwarding of IP Packets
+
+Since CONNECT-IP allows the transmission of IP packets over HTTP, CONNECT-IP
+endpoints will most often forward these packets to and from traditional IP
+interfaces. As such, CONNECT-IP endpoints act as IP routers. When a CONNECT-IP
+endpoint receives an HTTP Datagram containing an IP packet, it will parse the
+packet's IP header, perform any local policy checks (e.g., source address
+validation), check their routing table to pick an outbound interface, and then
+use an implementation-specific mechanism (such as raw sockets) to send the IP
+packet on that interface.
+
+Conversely, when a CONNECT-IP endpoint receives an IP packet whose destination
+address does not match any local addresses, it consults its routing table to
+pick a forwarding destination, and if the table points to a CONNECT-IP tunnel,
+the endpoint performs the same forwarding checks before transmitting the packet
+inside the tunnel.
+
+Note that CONNECT-IP endpoints will decrement the IP Hop Count (or TTL) upon
+encapsulation but not decapsulation. In other words, the Hop Count is
+decremented right before an IP packet is transmitted in an HTTP Datagram. This
+prevents infinite loops in the presence of routing loops, and matches the
+choices in IPsec {{?IPSEC=RFC4301}}.
```

Comment:
I'm not convinced this is the right strategy --- this requires clients, for example, to self-decrement the packet before they transmit it to their server. Usually TTLs are decremented when the routing decision is made, which implies that "upon encapsulation, but not decapsulation" is not the right heuristic. I think we want "decremented upon operations by the server terminating the CONNECT-IP connection"?

Reply:
That's reasonable, file an issue so we discuss this with the WG? I'd prefer to match IPsec for now